openedx/stanford/cms/envs/aws.py

```diff
@@ -13,6 +13,15 @@
     DEFAULT_COURSE_ABOUT_IMAGE_URL
 )
 EXTRA_MIMETYPES = ENV_TOKENS.get('EXTRA_MIMETYPES', EXTRA_MIMETYPES)
+FEATURES.update({
+    'ALLOW_COURSE_RERUNS': ENV_TOKENS.get(
+        'ALLOW_COURSE_RERUNS',
+        FEATURES.get(
+            'ALLOW_COURSE_RERUNS',
+            True
```

Comment:
Does this set the default to true solely for tests or is there some other purpose?

Reply:
Upstream, `settings.FEATURES['ALLOW_COURSE_RERUNS']` is always set to `True`.
This includes tests, production, everything.

On our fork, we were _disabling_ this value by default, but then re-enabling it for tests,
as the tests written by edx expect this to be the case (see above).

This change to the `stanford` settings is such that when the `aws`/`prod` settings are loaded, it will:
- try to pull the value out of `json`
- if not in `json`:
  - fall back to an already defined value (here, this implies `common.py`)
  - if not in `common.py`:
    - fall back to `True`

That said, the final fall back, `if not in common.py` is sort of a belt-and-suspenders check, as the code, as written, will always* be defined defined there (see: `cms/envs/common.py`). So, we could rewrite this such that it forcibly pulls the value from `common.py`, like:

```python
FEATURES.update({
    'ALLOW_COURSE_RERUNS': ENV_TOKENS.get(
        'ALLOW_COURSE_RERUNS',
        FEATURES['ALLOW_COURSE_RERUNS']
    ),
})
```

Tests don't load settings from `json`, so the effect here is that the feature is generally enabled everywhere,
unless a specific deployment chooses to disable it via JSON, which we're only doing on `edxapp1/2`.

Does that clarify?
And would you like me to simplify that logic as proposed?

Reply:
Yes, I think the simplified version is better. Thank you,